solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -1728,20 +1728,21 @@ private int fetchPackets(FastInputStream fis) throws Exception {
             long checkSumClient = checksum.getValue();
             if (checkSumClient != checkSumServer) {
               log.error("Checksum not matched between client and server for file: {}", fileName);
-              //if checksum is wrong it is a problem return for retry
+              //if checksum is wrong it is a problem return (there doesn't seem to be a retry in this case.)
               return 1;
             }
           }
           //if everything is fine, write down the packet to the file
           file.write(buf, packetSize);
           bytesDownloaded += packetSize;
           log.debug("Fetched and wrote {} bytes of file: {}", bytesDownloaded, fileName);
-          if (bytesDownloaded >= size)
-            return 0;
           //errorCount is always set to zero after a successful packet
           errorCount = 0;
+          if (bytesDownloaded >= size)
+            return 0;
         }
       } catch (ReplicationHandlerException e) {
+        log.warn("Aborting index replication", e);
```

Comment:
Is log-and-throw necessary here? it looks like we already log this higher up in the call stack.

Reply:
agreed. Logging here is not needed.